### PR TITLE
oprofile: depend on kernel headers to correctly compile 'operf' util.

### PR DESCRIPTION
**Pull request by ghost**
For 'operf' utility to be built linux kernel headers dependency should be added.
This is my first commit here, so please help me to be correct.

### CODE REVIEW

File: pkgs/development/tools/profiling/oprofile/default.nix [hunk 1]
```diff
@@ -1,4 +1,4 @@
-{ stdenv, fetchurl, binutils, popt, zlib, pkgconfig
+{ stdenv, fetchurl, binutils, popt, zlib, pkgconfig, linuxPackages
```
Comment (peti) [inline pkgs/development/tools/profiling/oprofile/default.nix]: Why do you take the entire package set as an argument if `kernelHeaders` is the only item from that set that you actually need?

File: pkgs/development/tools/profiling/oprofile/default.nix [hunk 2]
```diff
@@ -12,10 +12,14 @@ stdenv.mkDerivation rec {
     sha256 = "0nn4wfvwy4nii25y6lwlrnzx9ah4nz0r93yk7hswiy6wxjs10wc4";
   };
 
-  buildInputs = [ binutils zlib popt pkgconfig ]
+  buildInputs = [ binutils zlib popt pkgconfig linuxPackages.kernelHeaders ]
     ++ stdenv.lib.optionals withGUI [ qt4 ];
 
   configureFlags = [
+      "--build ${stdenv.system}"
```
Comment (peti) [inline pkgs/development/tools/profiling/oprofile/default.nix]: Those flags are necessary? Are you sure? I thought that Nix `system` ids don't correspond to those used by autoconf.

File: pkgs/development/tools/profiling/oprofile/default.nix (hunk 1 above)
Comment (tk-ecotelecom) [inline pkgs/development/tools/profiling/oprofile/default.nix]: Well, it's my first try to change anything in nixpkgs.
Actually I don't know how to pass only kernelHeaders here. Do you have any suggestion?

File: pkgs/development/tools/profiling/oprofile/default.nix (hunk 2 above)
Comment (tk-ecotelecom) [inline pkgs/development/tools/profiling/oprofile/default.nix]: I'm sure in only one thing: oprofile wouldn't not build if the target and host platform options are not specified when '--with-kernel' switch is used. I can check if build is necessary.

**Comment by tk-ecotelecom** [comment]
I've copied the dependency logic from glibc package. There is the same overriding. Why should I create something different?
OK. I'll create another commit...

**Comment by tk-ecotelecom** [comment]
~~overriding~~ :smile:

**Comment by vcunat** [comment]
The glibc expression would benefit from some refactoring, after all those years.

**Pull request merged**